### PR TITLE
fix(RHINENG-1695): GET /assignment-rules follow-ups

### PR DESCRIPTION
**Pull request by FabriciaDinizRH**
# Overview

This PR is being created to address [RHINENG-1695](https://issues.redhat.com/browse/RHINENG-1695).
It implements the following:
- fix RBAC attributeFilter check
- remove option to sort results by account and org_id
- add RBAC attributeFilter test
- remove assignments to unused variables
- remove manual url creation in tests
- rename `test_get_assignment_rule_with_bad_name`
- substitute "assignmentrule" for "assignment_rule" which is the name convention
- capitalize log message

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: tests/test_api_assignment_rules_get.py [hunk 1]
```diff
@@ -157,8 +157,10 @@ def test_assignment_rule_id_list_filter(num_rules, db_create_assignment_rule, db
     for idx in range(10):
         db_create_assignment_rule(f"extraRule_{idx}", group.id, filter, True)
 
-    url = ASSIGNMENT_RULE_URL + "/" + ",".join(assignment_rule_id_list)
-    response_status, response_data = api_get(url)
+    # url = ASSIGNMENT_RULE_URL + "/" + ",".join(assignment_rule_id_list)
```
Comment (kruai) [inline tests/test_api_assignment_rules_get.py:160]: This commented code should be removed

**Review by kruai — CHANGES_REQUESTED** [review]
Just one tiny suggestion but then it's good to go!

**Review by kruai — APPROVED** [review]

**Review by thearifismail — APPROVED** [review]
LGTM

**Pull request merged**